### PR TITLE
* SCP-2611 Correct gene models for dense matrices and account for single column feature files

### PR DESCRIPTION
**Pull request by knapii-developments**
There was a regression for parsing dense matrices where gene models were only created if the gene had expression data. The PR addresses this issue by creating gene models regardless of the presence of expression data. 

There was also a second regression. The feature file in an MTX bundle can have 1 column. As stated by [10X genomics](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/output/matrices), we now set the gene name to the gene ID in the event 1 column is present.

### CODE REVIEW

**Review by eweitz — APPROVED** [review]

**Review by devonbush — CHANGES_REQUESTED** [review]
Code looks good.  Please add a test case that confirms gene models get created for genes with no non-zero expression scores (i.e. a test case that would fail without this fix)

**Comment by knapii-developments** [comment]
> Code looks good. Please add a test case that confirms gene models get created for genes with no non-zero expression scores (i.e. a test case that would fail without this fix)

b197446

File: ingest/expression_files/mtx.py [hunk 1]
```diff
@@ -190,11 +190,21 @@ def transform(self):
...
-                    prev_gene_id, prev_gene = self.genes[prev_idx - 1].split("\t")
-                    # Ff the previous gene exists, load its models
+                    features = self.genes[prev_idx - 1].split("\t")
+                    prev_gene_id = features[0]
+                    try:
+                        prev_gene = features[1]
+                    # If no gene_name field is present, hence the index error,
+                    # gene name is equivalent to gene ID
+                    except IndexError:
+                        prev_gene = prev_gene_id
```
Comment (devonbush) [inline ingest/expression_files/mtx.py:206]: I think this will be more legible written as
```suggestion
                    if len(features) >= 2:
                        prev_gene = features[1]
                    # If no gene_name field is present,
                    # gene name is equivalent to gene ID
                    else:
                        prev_gene = prev_gene_id
```

**Review by devonbush — CHANGES_REQUESTED** [review]
Needs a test to cover the single-column gene file case.  And one minor code format suggestion

**Review by devonbush — APPROVED** [review]

**Review by eweitz — APPROVED** [review]

File: ingest/expression_files/dense_ingestor.py [hunk 2]
```diff
@@ -249,27 +249,19 @@ def transform(self):
...
                 data_arrays,
                 num_processed,
-                True,
+                False,
+            )
+        # Load any remaining models. This is necessary because the amount of
+        # models maybe less than the batch size.
+        if len(gene_models) > 0:
+            self.create_models(
+                [], [], None, None, gene_models, data_arrays, num_processed, True
```
Comment (jlchang) [inline ingest/expression_files/dense_ingestor.py:266]: Thanks for taking the time to explain this to me, Eno!

**Review by jlchang — APPROVED** [review]
